build/common/installer/scripts/tomlparser-agent-config.rb

```diff
@@ -273,6 +278,30 @@ def populateSettingValuesFromConfigMap(parsedConfig)
             @mdsdMonitoringMaxEventRate = mdsdMonitoringMaxEventRate.to_i
             puts "Using config map value: monitoring_max_event_rate  = #{@mdsdMonitoringMaxEventRate}"
           end
+          mdsdUploadMaxSizeInMB = mdsd_config[:upload_max_size_in_mb]
+          if is_valid_number?(mdsdUploadMaxSizeInMB)
+            @mdsdUploadMaxSizeInMB = mdsdUploadMaxSizeInMB.to_i
+            puts "Using config map value: upload_max_size_in_mb  = #{@mdsdUploadMaxSizeInMB}"
+          end
+          mdsdUploadFrequencyInSeconds = mdsd_config[:upload_frequency_seconds]
+          if is_valid_number?(mdsdUploadFrequencyInSeconds)
+            @mdsdUploadFrequencyInSeconds = mdsdUploadFrequencyInSeconds.to_i
+            puts "Using config map value: upload_frequency_seconds  = #{@mdsdUploadFrequencyInSeconds}"
+          end
+          mdsdBackPressureThresholdInMB = mdsd_config[:backpressure_memory_threshold_in_mb]
+          if is_valid_number?(mdsdBackPressureThresholdInMB) && mdsdBackPressureThresholdInMB.to_i > 100
+            @mdsdBackPressureThresholdInMB = mdsdBackPressureThresholdInMB.to_i
+            puts "Using config map value: backpressure_memory_threshold_in_mb  = #{@mdsdBackPressureThresholdInMB}"
+          else
+            puts "Ignoring mdsd backpressure limit. Check input values for correctness."
+          end
+          mdsdCompressionLevel = mdsd_config[:compression_level]
+          if is_valid_number?(mdsdCompressionLevel) && mdsdCompressionLevel.to_i < 10 # supported levels from 0 to 9
```

Comment:
can we add lowerbound check for compression level as well?

Reply:
> can we add lowerbound check for compression level as well?

> good catch. addressed. lower limit is 0 and upper limit is 9.